docker/pulsar/Dockerfile

```diff
@@ -33,7 +33,13 @@ COPY scripts/pulsar-zookeeper-ruok.sh /pulsar/bin
 COPY scripts/watch-znode.py /pulsar/bin
 COPY scripts/install-pulsar-client.sh /pulsar/bin
 
-RUN mkdir /pulsar/data
+# In order to support running this docker image as a container on OpenShift
+# the final image needs to give the root group sufficient permission.
+# The file permissions are preserved when copying files from this builder image to the target image.
+RUN chmod -R g+w /pulsar/conf
+RUN mkdir /pulsar/data && chmod -R g+w /pulsar/data
+RUN mkdir /pulsar/logs && chmod -R g+w /pulsar/logs
+RUN chmod -R u-w /pulsar
```

Comment:
When executing the instruction `RUN chmod -R u-w /pulsar`, Docker will record the history of all files in `/pulsar` as a extra layer, which may cause the image size to become larger. Could you provide a comparison of the image size before and after the changes?

Reply:
This is done in a builder image, so it does not affect the final docker image size.

Reply:
You can see in my docker repo [0] that the compressed size is currently 878.56 MB. In the official 2.9.0 image, the compressed size is 870.89 MB.

[0] https://hub.docker.com/layers/michaelmarshall/pulsar/2.10.0-SNAPSHOT/images/sha256-8b48a1a63da42023789e4ff02313117ff6fe81c7dff14cb0c9410c9d3e554158?context=explore
[1] https://hub.docker.com/layers/apachepulsar/pulsar/2.9.0/images/sha256-d056c89b7131979c681a437647b4c605594d7a95f338500c6a0cde6f6705df52?context=explore

Reply:
Update: after looking at this block more, I am going to remove `RUN chmod -R u-w /pulsar` because it is only removing write permissions on files owned by root. Note that the `RUN chmod -R g+w /pulsar/conf` is necessary and does not increase the overall size of the docker image.